tests/python/gaia-ui-tests/gaiatest/tests/functional/marketplace/test_marketplace_addon.py

```diff
@@ -0,0 +1,82 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import By, Wait
+from gaiatest import GaiaTestCase
+from gaiatest.apps.marketplace.app import MarketplaceDev
+from gaiatest.apps.settings.app import Settings
+from gaiatest.apps.system.app import System
+
+
+class TestSearchMarketplaceAndInstallAddon(GaiaTestCase):
+
+    addon_name_to_install = "'M' in Status Bar"
+    pref_certs = 'dom.mozApps.use_reviewer_certs'
+    pref_install = 'dom.mozApps.signed_apps_installable_from'
+    _status_bar_m_locator = (By.CSS_SELECTOR, '#statusbar-maximized [id="statusbar-m"]')
+
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+        self.data_layer.set_bool_pref(self.pref_certs, True)
+        self.data_layer.set_char_pref(self.pref_install,
+            'https://marketplace.firefox.com,https://marketplace-dev.allizom.org')
+        self.connect_to_local_area_network()
+
+    def test_search_and_install_app(self):
+
+        marketplace = MarketplaceDev(self.marionette)
+        marketplace.launch()
+
+        results = marketplace.install_addon(self.addon_name_to_install)
+
+        settings = Settings(self.marionette)
+        settings.launch()
+        addons_page = settings.open_addons()
+
+        for x in addons_page.items:
+            if x.addon_name == self.addon_name_to_install:
+                addon_details = x.tap()
+                break
+
+        self.assertTrue(addon_details.is_enabled())
+
+        self.assertEquals(addon_details.description, "Adds a 'M' to the status bar")
+        self.assertEquals(self.apps.is_app_installed(addon_details.affected_apps)['name'], 'System')
+        self.apps.switch_to_displayed_app()
+
+        system = System(self.marionette)
+        status_bar = system.status_bar
+        elements = status_bar.root_element.find_elements(*self._status_bar_m_locator)
+
+        self.assertEquals(len(elements), 1)
+
+        # is_displayed() returns False for some reason
+        element = elements[0]
+        self.assertNotEquals(element.rect['x'], 0)
+        self.assertNotEquals(element.rect['y'], 0)
+        self.assertNotEquals(element.rect['width'], 0)
+        self.assertNotEquals(element.rect['height'], 0)
+        # element.text doesn't work, because Marionette thinks the element is not displayed for some reason
+        self.assertEquals(self.marionette.execute_script('return arguments[0].innerHTML', [element]), 'M')
+
+        self.apps.switch_to_displayed_app()
+        addon_details.refresh_root_element()
```

Comment:
What's the value of refreshing the element in the test? In my opinion, this should be an implementation details, that should belong in the Page Class.
What would it take to put that call in `disable()`?